buildconfig/pygame-stubs/_common.pyi

```diff
@@ -31,4 +31,7 @@ _CanBeRect = Union[
 class _HasRectAttribute(Protocol):
     rect: _CanBeRect
 
-_RectValue = Union[_CanBeRect, _HasRectAttribute]
+class _HasCallableRectAttribute(Protocol):
```

Comment:
Is this the right way to express a type hint for all the possible ways the collision methods can accept?

Reply:
I'd suggest leaving out this typing fix in this PR, I've already fixed this in #2999 a few days back :)
your fix will conflict with my fix

Reply:
yes, especially since you have moved the file to a different location if I have interpreted your pr correctly.

Reply:
Yeah. And the fix I make is the same fix made here

Reply:
reverted the changes except the typo... should I revert the changes in rect.pyi too?